newsfragments/3899.changed.md

```diff
@@ -0,0 +1,2 @@
+The `wrap_pyfunction!` macro is now deprecated. The `wrap_pyfunction_bound!` macro, which returns a `Bound<'_, PyCFunction>` should be used instead.
```

Comment:
We might not actually need this bit, because it's part of the GIL Ref refactor.

Reply:
Yeah, we skipped all the other gil-ref refactorings from the changelog, so I don't think we should start that here now.

The change to the `#[pymodule]` macro should be mentioned, but as in the other comment, I think we should land that separately.

Reply:
I will split off the pymodule macro changes, but I'm not exactly sure what your suggestion is for this news fragment. Are you suggesting I keep the notice about deprecating `wrap_pyfunction!` but omit the reference to `Bound`, or should I not add a newsfragment at all?

Reply:
I think you can remove the fragment here, and then add one with the notice about the `#[pymodule]` change in that new PR.